### PR TITLE
Fix Windows build & CI

### PR DESCRIPTION
**Pull request by feedmeapples**
## What was changed

Fixed Windows build, Windows CI and lint issues, `trace` command on Windows, build e2e server locally instead of using the latest release.. etc CI related

## Why?

green checks

resolve https://github.com/temporalio/cli/issues/173
resolve https://github.com/temporalio/cli/issues/268

## Checklist

1. Closes 

2. How was this tested:

CI

3. Any docs updates needed?

### CODE REVIEW

File: trace/term_size_windows.go
Comment (bergundy) [inline trace/term_size_windows.go:43]: Does this return anything that can be used to determine if it succeeds?

**Review by bergundy — APPROVED** [review]

File: trace/term_size_windows.go
Comment (cretz) [inline trace/term_size_windows.go:1]: Ug, I was a bit disappointed to see the trace implementation use manual ANSI codes and syscalls everywhere. I wish we had encouraged https://github.com/gdamore/tcell or similar.

Is this change just to fix compilation or can you confirm the trace command works well on Windows? If you can confirm, no prob. If you're just fixing compilation but it's still broken/wrong at runtime, I think you should just make this entire command only available on nix builds and put a disclaimer somewhere on why.

Comment (feedmeapples) [reply trace/term_size_windows.go:1]: i'm testing this on Windows. Yeah ideally we wouldn't do platform specific coding

Comment (feedmeapples) [reply trace/term_size_windows.go:1]: i'll see before merging this PR if we can remove platform specific code

File: trace/term_size_windows.go
Comment (feedmeapples) [reply trace/term_size_windows.go]: it does, added a check

File: trace/term_size_windows.go
Comment (cretz) [reply trace/term_size_windows.go:1]: I don't mind the small platform-specific parts (e.g. sigint sending), I just want to make sure this trace feature works on all platforms we're making it available for

**Review by cretz — CHANGES_REQUESTED** [review]

**Comment by feedmeapples** [comment]
still dealing with this failure https://github.com/temporalio/cli/actions/runs/5262862061/jobs/9512447848#step:5:154

`go test` succeeds when running locally on Windows, however not in Github Actions. Haven't found yet what causes this

**Comment by feedmeapples** [comment]
sent one more blocker PR into Go SDK https://github.com/temporalio/sdk-go/pull/1140

File: tests/e2e_test.go
Comment (feedmeapples) [inline tests/e2e_test.go:43]:
```suggestion
	// ensure all servers are stopped in case not explicitly stopped from tests
```

**Review by samanbarghi — APPROVED** [review]

File: go.mod
Comment (cretz) [inline go.mod:22]: Let's wait for server tag

**Review by cretz — CHANGES_REQUESTED** [review]

**Review by cretz — APPROVED** [review]
Approving with the guarantee that we are not going to tag CLI with an untagged server dependency

**Pull request merged**
